main-app/app/scripts/providers/game-model.js

```diff
@@ -0,0 +1,126 @@
+                me.gameOver = true; //possibly not actually used anywhere, but its set at the right times so might be useful
+                me.showWinner = false;
+                me.winnerText = "";
+
+                me.isPlayerTypeValid = function(playerNum){
+                    if(playerNum===1){
+                        for (var i = 0; i < me.playerTypes.length; i++) {
+                            if (me.playerTypes[i].value === me.player1Type) {
+                                return true;
+                            }
+                        }
+                    }
+                    else if(playerNum===2){
+                        for (var j = 0; j < me.playerTypes.length; j++) {
+                            if (me.playerTypes[j].value === me.player2Type) {
+                                return true;
+                            }
+                        }
+                    }
+                    return false;
+                };
+
+                me.setupBoard = function (player1Type, player2Type, player1Icon) {
+                    if (player1Icon == "Crosses") {
+                        me.playerCross = 1;
+                        me.playerNought = 2;
+                    }
+                    else {
+                        me.playerCross = 2;
+                        me.playerNought = 1;
+                    }
+
+                    proxy.setupBoard({player1: player1Type, player2: player2Type})
+                        .success(function (data, status, headers, config) {
+                            $location.path("/game");
+                            me.showWinner = false;
+                            $timeout.cancel();
+                            update(data);
+                            audio.playButtonPress();
+                        })
+                        .error(function (data, status, headers, config) {
+                            alert(data);
+                        });
+
+                    me.gameOver = false;
+                };
+
+                me.makeMove = function (choice) {
+                    proxy.makeMove({"playerNumber": 1, "chosenSquare": choice})
+                        .success(function (data, status, headers, config) {
+                            update(data);
+                            audio.playMove();
+                        })
+                        .error(function (data, status, headers, config) {
+                            alert(data);
+                        });
+                };
+
+                var update = function (data) {
+                        me.currentGrid = boardToArray(data.gameboard);
+                        if (data.outcome !== "Continue") {
+                            endGame(data);
+                        }
+                        console.log(data.gameboard);
+                        console.log(data.outcome);
+                        console.log(data.winner);
+                    },
+
+                    boardToArray = function (board) {
```

Comment:
You can move this out to a provider under the proxy module - 'Tombola.NoughtsAndCrosses.Providers.Proxy'
call it something like gameBoardFactory, it's normal to want to process the [deliberately] only-just-good-enough repsonse from the server into something more usable in the front-end in any system